externalId-migration-util/src/main/java/org/sunbird/migration/RecordProcessor.java

```diff
@@ -0,0 +1,249 @@
+    }
+
+    /**
+     * this method should be used to get the instance of the class..
+     *
+     * @param connectionFactory
+     * @param requestParams
+     * @return
+     */
+    public static RecordProcessor getInstance(
+            ConnectionFactory connectionFactory, RequestParams requestParams) {
+        return new RecordProcessor(connectionFactory, requestParams);
+    }
+
+
+    /**
+     * this method is responsible to fetch the user external id data from cassandra and the table name
+     * is usr_external_identity and after fetching will convert the resultSet to List<User>
+     *
+     * @return List<User>
+     */
+    private Map<String,String> getOrgDataFromDbAsList() {
+        ResultSet resultSet = connection.getRecords("select * from organisation");
+        Map<String,String> orgProviderMap = CassandraHelper.getOrgProviderMapFromResultSet(resultSet);
+        logger.debug(orgProviderMap.entrySet().iterator().next());
+        return orgProviderMap;
+    }
+
+    /**
+     * this method is responsible to fetch the user external id data from cassandra and the table name
+     * is usr_external_identity and after fetching will convert the resultSet to List<User>
+     *
+     * @return List<User>
+     */
+    private List<User> getUserDataFromDbAsList() {
+        ResultSet resultSet = connection.getRecords("select * from usr_external_identity");
+        List<User> usersList = CassandraHelper.getUserListFromResultSet(resultSet);
+        logger.debug(usersList.iterator().next());
+        return usersList;
+    }
+
+    /**
+     * this method will be used to process externalId this method will get the count of total number
+     * of records that need to be processed
+     *
+     * @return
+     */
+    public void startProcessingExternalIds() throws IOException {
+        List<User> usersList = getRecordsToBeProcessed();
+        Map<String,String> orgIdProviderMap = getOrgDataFromDbAsList();
+        Map<String,List<User>> userIdExternalIdMap = createExternalInfoMapGroupbyUserId(usersList);
+
+        List<UserDeclareEntity> userSelfDeclareLists = new ArrayList<>();
+        getUserSelfDeclareLists(userIdExternalIdMap, userSelfDeclareLists);
+        userSelfDeclareLists
+                .stream()
+                .forEach(
+                        userSelfDeclareObject->{
+                            try{
+                                startTracingRecord(userSelfDeclareObject.getUserId());
+                                performSequentialOperationOnRecord(userSelfDeclareObject,orgIdProviderMap);
+                            }catch (Exception ex){
+                                logExceptionOnProcessingRecord(userSelfDeclareObject.getUserId(),userSelfDeclareObject.getProvider(),
+                                        userSelfDeclareObject.getPersona());
+                            }finally {
+                                endTracingRecord(userSelfDeclareObject.getUserId());
+                            }
```

Comment:
Can we think of helping ourselves with some metrics?
1. We had n many rows in existing table.
2. We now have n - X many rows now (because we moved).
3. X many rows are there in the new table.

Think of it as incrementing some counters - not necessarily as count() aggregate db function.

Reply:
Done